libbeat/processors/condition.go

```diff
@@ -259,12 +267,22 @@ func (c *Condition) checkEquals(event ValuesMap) bool {
 			}
 		} else {
 			sValue, err := extractString(value)
-			if err != nil {
-				logp.Warn("unexpected type %T in equals condition as it accepts only integers and strings. ", value)
-				return false
-			}
-			if sValue != equalValue.Str {
-				return false
+
+			if err == nil {
```

Comment:
Same comment as above, use continue to make the code more readable and less nested.